addons/mail/models/mail_thread.py

```diff
@@ -4507,9 +4507,14 @@ def _message_update_content(self, message, body, attachment_ids=None, partner_id
         if strict:
             self._check_can_update_message_content(message.sudo())
 
-        msg_values = {
-            'body': escape(body),  # keep html if already Markup, otherwise escape
-        } if body is not None else {}
+        msg_values = {}
+        if body is not None:
+            msg_values["body"] = (
+                # keep html if already Markup, otherwise escape
+                escape(body) + Markup("<span class='o-mail-Message-edited'/>")
+                if body or attachment_ids or (attachment_ids is None and message.attachment_ids)
```

Comment:
Why add condition on attachments actually? I thought we were good with just having it depend on `body is not None` and only add the label for body changes.

Reply:
If you need to detect message empty you should use `_filter_empty` or something similar (or extract the logic to a function you can call) because it includes more conditions than this.

Reply:
When there is no attachment, delete means delete. But when there is some attachments, delete means empty body but with an edited label.

Reply:
like this
![4440379027](https://github.com/user-attachments/assets/466c79b5-4370-4c25-a41e-28cf66437ece)